src/index.js

```diff
@@ -100,8 +101,40 @@ const middleware = (config = {}) => (req, res, next) => {
         error: 'Signature verification failed!'
       })
     }
+    const certificates = JSON.parse(req.headers['x-authrite-certificates']).map(async cert => {
+      if(cert.subject !== identityKey) {
+        return res.status(401).json({
+          error: 'Certificate subject does not match identity key of the request sender!'
+        })
+      }
+
+      // Check valid signature
+      try {
+        authriteUtils.verifyCertificateSignature(cert)
+      } catch (e) {
+        return res.status(401).json({
+          error: `Invalid certificate signature: ${cert.signature}`
+        })
+      }
+
+      // Check encrypted fields and decrypt them
+      let decryptedFields = []
```

Comment:
```suggestion
      let decryptedFields = {}
```
The return value from this function is an object.